src/Client.php

```diff
@@ -352,7 +352,9 @@ public function send($method, $payload, $path, $params = [], array $headers = []
         $response = call_user_func($this->middleware, $request, $response, $this);
 
         if ($response->getStatusCode() === 404) {
-            throw new Http\Exception\NotFoundException();
+            $content = json_decode($response->getBody()->getContents(), true);
+
+            throw new Http\Exception\NotFoundException($content['message'] ?? '');
```

Comment:
I think this is incorrect. The exception message is in the HTTP Status reason

Reply:
`$response->getReasonPhrase()` returns 'Not Found'. I need detailed error: 'Customer not found'. How can I achieve that?

Reply:
Improve message on backend?